src/VTK/VtkGrid2DRepresentationCells.cxx

```diff
@@ -55,10 +66,13 @@ VtkGrid2DRepresentationCells::~VtkGrid2DRepresentationCells()
 //----------------------------------------------------------------------------
 void VtkGrid2DRepresentationCells::createOutput(const std::string & uuid)
 {
+#ifdef WITH_TEST
+	BEGIN_FUNC(__func__);
+#endif
 	if (!subRepresentation)	{
 
 		resqml2_0_1::Grid2dRepresentation* grid2dRepresentation = nullptr;
-		common::AbstractObject* obj = epcPackageRepresentation->getResqmlAbstractObjectByUuid(uuid);
+		common::AbstractObject* obj = epcPackageRepresentation->getDataObjectByUuid(uuid);

```

Comment:
you can use the template version of getDataObjectByUuid which automatically try to cast into Grid2dRepresentation.
You can then jsut check nullity at line below.